packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-has-one-relationship/schema.ts

```diff
@@ -0,0 +1,193 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+import { StudioForm } from '@aws-amplify/codegen-ui';
+import { Schema } from '@aws-amplify/datastore';
+
+export const bookUpdateForm: StudioForm = {
+  name: 'BookUpdateForm',
+  dataType: { dataSourceType: 'DataStore', dataTypeName: 'Book' },
+  formActionType: 'update',
+  fields: {
+    primaryAuthor: {
+      inputType: {
+        type: 'Autocomplete',
+        valueMappings: {
+          values: [
+            {
+              value: {
+                bindingProperties: {
+                  property: 'Author',
+                  field: 'id',
+                },
+              },
+              displayValue: {
+                bindingProperties: {
+                  property: 'Author',
+                  field: 'name',
+                },
+              },
+            },
+          ],
```

Comment:
Maybe a dumb question. Could you provide me with the context about how do we get this form definition. specifically this `valueMappings`?

Reply:
The `fields` part of _StudioForm_ definition defines what the form fields will be.  In this case, it will be an `AutoComplete` field with bound properties from the Author model.  In the case of a golden file, I'm not sure how useful it is to define a `StudioForm` object it.  The `Schema` definition listed below is more useful since it defines the relationship schema for this golden file.